roles/edpm_reboot/defaults/main.yaml

```diff
@@ -0,0 +1,21 @@
+---
+# Copyright 2023 Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+
+# All variables intended for modification should be placed in this file.
+
+edpm_reboot_workload_protection: true
```

Comment:
Instead of having a variable to control this, I think we should save some state on the node itself. That way, we can be sure to reboot on initial deployment, when it is most likely needed after configuring kernel args.

This was done in TripleO by checking for the existence of the generated nova configuration files. If those files existed, we never automatically rebooted unless it was forced.

This could work the same way by writing out a file somewhere (/var/lib/openstack/initial_reboot_completed or similar). If that file exists, we never reboot unless a var is set to override it.

This also makes reboots work out of the box on initial deployment without the user having to remember to toggle this variable to false, then back to true.

Reply:
I changed the code logic to run reboot on initial deployment always if reboot is required.
However I had to left variable, because otherwise after initial deployment reboot would to be skipped always if not forced. New variable is edpm_reboot_force_reboot set to false by default.